### PR TITLE
Clean games dashboard edge cases

### PR DESCRIPTION
**Pull request by AlexandreSi**
- Reset games when user logs out
- Correctly highlight currently opened project in games dashboard

### CODE REVIEW

File: newIDE/app/src/GameDashboard/GamesList.js [hunk 1]
```diff
@@ -629,8 +629,11 @@ const GamesList = ({
                 const isCurrentProjectOpened =
                   (!!projectUuid && (!!game && game.id === projectUuid)) ||
                   (!!projectFileMetadataAndStorageProviderName &&
+                    !!currentFileMetadata &&
                     projectFileMetadataAndStorageProviderName.fileMetadata
-                      .gameId === projectUuid);
+                      .gameId === projectUuid &&
+                    projectFileMetadataAndStorageProviderName.fileMetadata
+                      .fileIdentifier === currentFileMetadata.fileIdentifier);
```
Comment (ClementPasteau) [inline newIDE/app/src/GameDashboard/GamesList.js:636]: what issue is this solving? Not 100% clear in which case this happens

File: newIDE/app/src/GameDashboard/UseGamesList.js [hunk 2]
```diff
@@ -31,6 +31,14 @@ const useGamesList = (): GamesList => {
     null
   );
 
+  React.useEffect(
+    () => {
+      // Reset games when user logs out.
```
Comment (ClementPasteau) [inline newIDE/app/src/GameDashboard/UseGamesList.js:36]: When I'm logging out, I don't see my games any more 🤔 why is this needed?

Comment (AlexandreSi) [reply newIDE/app/src/GameDashboard/UseGamesList.js:36]: I was in a case, when the games were still available. I will try to reproduce it

Comment (AlexandreSi) [reply newIDE/app/src/GameDashboard/UseGamesList.js:36]: I don't reproduce it, I'll get rid of it

**Review by ClementPasteau — APPROVED** [review]
